docs/figma-code-connect.md

```diff
@@ -0,0 +1,174 @@
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Prerequisites](#prerequisites)
+- [Setup](#setup)
+- [Creating Code Connect Files](#creating-code-connect-files)
+- [Publishing Components](#publishing-components)
+- [Best Practices](#best-practices)
+- [Resources](#resources)
+
+## Overview
+
+Figma Code Connect bridges the gap between design and development by:
+
+- **Displaying Real Code**: Shows actual component usage in Figma Dev Mode
+- **Multi-Platform Support**: Supports both React and React Native with platform-specific labels
+- **Automatic Prop Mapping**: Maps Figma component properties to code component props
+- **Live Updates**: Keeps code examples in sync with our component implementations
+
+## Prerequisites
+
+### 1. Figma Personal Access Token
+
+Generate a personal access token from [Figma Developer Settings](https://www.figma.com/developers/api#access-tokens) with the following scopes:
+
+- **Code Connect (Write)**: Required for publishing components
+- **File content (Read-only)**: Required for reading component definitions
+
+### 2. Environment Setup
+
+Copy the example environment file and add your token:
+
+```bash
+cp .env.example .env
+# Edit .env and add your Figma access token
+```
+
+## Setup
+
+### 1. Monorepo Configuration
+
+Our monorepo uses individual `figma.config.json` files for each package with root-relative paths:
+
+- `@metamask/design-system-react`: [packages/design-system-react/figma.config.json](packages/design-system-react/figma.config.json)
+- `@metamask/design-system-react-native`: [packages/design-system-react-native/figma.config.json](packages/design-system-react-native/figma.config.json)
+
+The `@figma/code-connect` dependency is managed at the root level and used by both packages through the monorepo scripts.
+
+### 2. Monorepo Scripts
+
+Use these commands from the **root directory**:
+
+```bash
+# Interactive setup wizards (for adding/updating components)
+yarn figma:connect:react              # Interactive setup for React components
+yarn figma:connect:react-native       # Interactive setup for React Native components
+
+# Automated publishing (for CI/CD workflows)
+yarn figma:connect:publish            # Publish all components (React + React Native)
+yarn figma:connect:publish:dry-run    # Dry run to validate all components
+yarn figma:connect:unpublish          # Unpublish all components
+
+# Platform-specific automated publishing
+yarn figma:connect:publish:react              # Publish only React components
+yarn figma:connect:publish:react-native       # Publish only React Native components
+yarn figma:connect:publish:react:dry-run      # Dry run for React components
+yarn figma:connect:publish:react-native:dry-run  # Dry run for React Native components
+
+# Platform-specific unpublishing
+yarn figma:connect:unpublish:react           # Unpublish React components
+yarn figma:connect:unpublish:react-native    # Unpublish React Native components
```

Comment:
Publishing and un-publishing components. The top 3 scripts are for both React and React Native so we only need to run the first three to test all 9 Platform-specific scripts.

https://github.com/user-attachments/assets/dc9f388c-3005-4fe9-8a78-5f90c007438c